src/deadline/job_attachments/download.py

```diff
@@ -372,48 +372,20 @@ def download_files_in_directory(
     )
 
 
-def download_file(
-    file: RelativeFilePath,
-    hash_algorithm: HashAlgorithm,
-    local_download_dir: str,
+def download_file_with_s3_key(
```

Comment:
Could you add test for the new function?

Reply:
with `download_file_with_s3_key`, all of the functionality was purely refactored out of `download_file`, since I wanted to use only part of the function in the asset CLI. There are already tests for the `download_file` module, which tests the same functionalities would I need to add additional ones ?

Reply:
It would be nice to have the coverage for `download_file_with_s3_key`. For example, I don't think the signature of this function is tested.